src/App.js

```diff
@@ -1,18 +1,46 @@
 import React, { Component } from 'react';
-import logo from './logo.svg';
+import AddTask from './AddTask';
+import TaskList from './TaskList';
 import './App.css';
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      inputValue: '',
+      tasks: []
+    };
+  }
+
+  handleChange(value) {
+    this.setState({ inputValue: value });
+  }
+
+  addTask() {
+    var tasks = this.state.tasks.slice();
+    tasks.push(this.state.inputValue);
+    this.setState({
+      tasks: tasks,
+      inputValue: ''
+    });
+  }
+
   render() {
+    var title = 'Git R Dun';
+
     return (
       <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
-        </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
+        <h1>{title}</h1>
+      
+        <input
+          value={this.state.inputValue}
+          onChange={e => this.handleChange(e.target.value)}
```

Comment:
Last night, you saw I had to add `.bind(this)` to the constructor to have access to `this`. I recommend reading about it [here](https://reactjs.org/docs/handling-events.html). In the above example, instead I'm using an arrow function directly on the `onChange` which will also bind the correct `this`. However, the linked article suggests to be careful with approach because it will cause a callback to be created each time the component renders.

Reply:
Wow, I was just reading more about event handling in react. I thought [this article](https://medium.com/@pauloesteves8/es6-classes-binding-public-class-fields-and-event-handling-in-react-2e1e39b1d498) did a good job of explaining all the options and differences.